web/react/components/textbox.jsx

```diff
@@ -27,11 +40,48 @@ module.exports = React.createClass({
     },
     componentWillUnmount: function() {
         PostStore.removeAddMentionListener(this._onChange);
+        ErrorStore.removeChangeListener(this._onError);
     },
     _onChange: function(id, username) {
         if (id !== this.props.id) return;
         this.addMention(username);
     },
+    _onError: function() {
+        var errorState = getStateFromStores();
+
+        if (this.state.timerInterrupt != null) {
+            window.clearInterval(this.state.timerInterrupt);
+            this.setState({ timerInterrupt: null });
+        }
+
+        if (errorState.message === "There appears to be a problem with your internet connection") {
+            this.setState({ connection: " bad-connection" });
```

Comment:
Yeah sure, it's done in a similar fashion in other parts of the code but doing it when it's part of the state does look a little extra odd.